biddingApi/biddingApi/src/test/java/biddingApi/biddingApi/ApiTestRestAssured.java

```
@@ -418,57 +261,121 @@ public void getTruckDataWithParam_LoadId() throws Exception {
+
+	}
+
+	@Test
+	@Order(12)
+	public void updateFailed_Shipper_Cannot_Update_TruckID() throws Exception {
+
+		BidPutRequest bidPutRequest = new BidPutRequest((long) 1000, BiddingData.Unit.PER_TRUCK,
+				Arrays.asList("truckID:abcd"), null, true, null);
+
+		String inputJson = mapToJson(bidPutRequest);
+
+		Response response = RestAssured.given().header("", "").body(inputJson).header("accept", "application/json")
+				.header("Content-Type", "application/json").put("/" + id1).then().extract().response();
+
+		assertEquals(200, response.statusCode());
+		assertEquals(Constants.TRUCK_ID_UPDATE_BY_SHIPPER, response.jsonPath().getString("status"));
+
+	}
+
+	@Test
+	@Order(13)
+	public void updateFailed_Transporter_Approval_Shipper_Approval_both_notNull() throws Exception {
+
+		BidPutRequest bidPutRequest = new BidPutRequest((long) 1000, BiddingData.Unit.PER_TRUCK,
+				Arrays.asList("truckID:abcd"), true, true, null);
+
+		String inputJson = mapToJson(bidPutRequest);
+
+		Response response = RestAssured.given().header("", "").body(inputJson).header("accept", "application/json")
+				.header("Content-Type", "application/json").put("/" + id1).then().extract().response();
+
+		assertEquals(200, response.statusCode());
+		assertEquals(Constants.TRANSPORTER_SHIPPER_APPROVAL_NOT_NULL, response.jsonPath().getString("status"));
+
+	}
+
+	@Test
+	@Order(14)
+	public void updateFailed_Account_Not_Found() throws Exception {
+
+		BidPutRequest bidPutRequest = new BidPutRequest((long) 1000, BiddingData.Unit.PER_TRUCK,
+				Arrays.asList("truckID:abcd"), true, null, null);
+
+		String inputJson = mapToJson(bidPutRequest);
+
+		Response response = RestAssured.given().header("", "").body(inputJson).header("accept", "application/json")
+				.header("Content-Type", "application/json").put("/" + "bid:abcd").then().extract().response();
+
+		assertEquals(200, response.statusCode());
+		assertEquals(Constants.uDataNotExists, response.jsonPath().getString("status"));
+
+	}
```

write a case where transporterApproval == false or shiperApproval == false, then we should return an error.